File: generators/hittiteNames.py
__all__ = ['hittiteNames']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nameGen'])
@Js
def PyJsHoisted_nameGen_(this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this}, var)
    var.registers(['nm1', 'result'])
    var.put('nm1', Js([Js('Abbi-Teshub'), Js('Akhsheri'), Js('Aki-Teshup'), Js('Akia'), Js('Alakshandu'), Js('Alarandu'), Js('Ambaris'), Js('Ammikhatna'), Js('Anittas'), Js('Anuwanza'), Js('Arame'), Js('Arandas'), Js('Arandash'), Js('Arnuanta'), Js('Arnuwandas'), Js('Artatamash'), Js('Asiri'), Js('Aulkia'), Js('Bakhianu'), Js('Bit-Burutish'), Js('Biyassili'), Js('Burutash'), Js('Dudkaliya'), Js('Dudkhalia'), Js('Dushratta'), Js('Erisinni'), Js('Esarhaddon'), Js('Etaqama'), Js('Gil-Teshup'), Js('Giliyash'), Js('Gunzinanu'), Js('Hasammeli'), Js('Hattusilis'), Js('Hishimisharruma'), Js('Ishganshar'), Js('Ishtanu'), Js('Kali-Teshup'), Js('Karparune'), Js('Katuzili'), Js('Khalpashshulubis'), Js('Khapashshubilish'), Js('Khattasulis'), Js('Khattushar'), Js('Khulli'), Js('Kielranu'), Js('Kil-Teshup'), Js('Kili-Teshup'), Js('Kilundu'), Js('Kundashpi'), Js('Kurirpa'), Js('Kuruntash'), Js('Kushtashpi'), Js('Labarnas'), Js('Lubarna'), Js('Lutipri'), Js('Manapa-Tarhunda'), Js('Manapa-Teshup'), Js('Murshil'), Js('Murshilish'), Js('Mursilis'), Js('Mutallu'), Js('Muttallu'), Js('Muwatalis'), Js('Muwatallis'), Js('Nabushezibanni'), Js('Pappash'), Js('Pikkandu'), Js('Pilandu'), Js('Pisandu'), Js('Pisiris'), Js('Pisirish'), Js('Piyamaradus'), Js('Purutash'), Js('Qatazilu'), Js('Sangar'), Js('Sapalulme'), Js('Sardur'), Js('Shadi-Teshup'), Js('Shalmaneser'), Js('Shama-Teshup'), Js('Sharkenkate-Ashira'), Js('Shaushkash'), Js('Shipa'), Js('Shirindu'), Js('Shubbiluliuma'), Js('Sin-Teshupash'), Js('Suppiliamus'), Js('Suppiliumash'), Js('Suppiluliuma'), Js('Surash'), Js('Surri'), Js('Sutarna'), Js('Sutatarra'), Js('Tarhu'), Js('Tarhunnas'), Js('Tarkhundapi'), Js('Tarkhundaraba'), Js('Tarkhundarabush'), Js('Tarkhuntash'), Js('Tarkondemos'), Js('Tashshu-Dasha'), Js('Te-Teshub'), Js('Telipinus'), Js('Teshub'), Js('Teti'), Js('Tudhaliuas'), Js('Tudhaliyas'), Js('Tudkhalias'), Js('Tudkhaliya'), Js('Tushratta'), Js('Ualli'), Js('Uassurme'), Js('Ullusunu'), Js('Urhiteshub'), Js('Uriah'), Js('Urikki'), Js('Urimme'), Js('Ushkhitti'), Js('Zananzash'), Js('Zannanza'), Js('Zurashar')]))
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm1').get('length'))|Js(0.0)))
            var.put('names', var.get('nm1').get(var.get('rnd')))
            var.get('nm1').callprop('splice', var.get('rnd'), Js(1.0))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
pass
pass


# Add lib to the module scope
hittiteNames = var.to_python()

File: js/hittiteNames.js
function nameGen(){
	var nm1 = ["Abbi-Teshub","Akhsheri","Aki-Teshup","Akia","Alakshandu","Alarandu","Ambaris","Ammikhatna","Anittas","Anuwanza","Arame","Arandas","Arandash","Arnuanta","Arnuwandas","Artatamash","Asiri","Aulkia","Bakhianu","Bit-Burutish","Biyassili","Burutash","Dudkaliya","Dudkhalia","Dushratta","Erisinni","Esarhaddon","Etaqama","Gil-Teshup","Giliyash","Gunzinanu","Hasammeli","Hattusilis","Hishimisharruma","Ishganshar","Ishtanu","Kali-Teshup","Karparune","Katuzili","Khalpashshulubis","Khapashshubilish","Khattasulis","Khattushar","Khulli","Kielranu","Kil-Teshup","Kili-Teshup","Kilundu","Kundashpi","Kurirpa","Kuruntash","Kushtashpi","Labarnas","Lubarna","Lutipri","Manapa-Tarhunda","Manapa-Teshup","Murshil","Murshilish","Mursilis","Mutallu","Muttallu","Muwatalis","Muwatallis","Nabushezibanni","Pappash","Pikkandu","Pilandu","Pisandu","Pisiris","Pisirish","Piyamaradus","Purutash","Qatazilu","Sangar","Sapalulme","Sardur","Shadi-Teshup","Shalmaneser","Shama-Teshup","Sharkenkate-Ashira","Shaushkash","Shipa","Shirindu","Shubbiluliuma","Sin-Teshupash","Suppiliamus","Suppiliumash","Suppiluliuma","Surash","Surri","Sutarna","Sutatarra","Tarhu","Tarhunnas","Tarkhundapi","Tarkhundaraba","Tarkhundarabush","Tarkhuntash","Tarkondemos","Tashshu-Dasha","Te-Teshub","Telipinus","Teshub","Teti","Tudhaliuas","Tudhaliyas","Tudkhalias","Tudkhaliya","Tushratta","Ualli","Uassurme","Ullusunu","Urhiteshub","Uriah","Urikki","Urimme","Ushkhitti","Zananzash","Zannanza","Zurashar"];

	var br = "";
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		rnd = Math.random() * nm1.length | 0;
		names = nm1[rnd];
		nm1.splice(rnd, 1);
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}